### PR TITLE
[FLUSH-3] PLU-65 feat(archival): add buildS3Key

### PR DESCRIPTION
**Pull request by kevinkim-ogp**
## TL;DR

Adds `buildS3Key` — a pure function that constructs the S3 object key for a given execution. Produces a Hive-style partitioned path used by the archival writer in later PRs.

## What changed?

**`src/helpers/archival/build-s3-key.ts`** — `buildS3Key(execution)` builds a key of the form:

```
executions/flow_id=<flowId>/year=<YYYY>/month=<MM>/execution_id=<id>.json.gz
test-executions/flow_id=<flowId>/year=<YYYY>/month=<MM>/execution_id=<id>.json.gz
```

- `S3_PREFIX_EXECUTIONS` / `S3_PREFIX_TEST_EXECUTIONS` constants exported for use in tests and callers.
- Partitions by UTC year/month; month is zero-padded.
- `createdAt` typed as `string` — matches the ISO string Knex/pg returns from the DB.

**`src/helpers/archival/build-s3-key.test.ts`** — unit tests covering the executions prefix, test-executions prefix, and single-digit month zero-padding.

## Tests

- [ ] `npm run test packages/backend/src/helpers/archival/build-s3-key.test.ts` passes

### CODE REVIEW

**Comment by kevinkim-ogp** [comment]
* **#1727** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1727?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#1728](https://github.com/opengovsg/plumber/pull/1728) <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1728?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1742](https://github.com/opengovsg/plumber/pull/1742) <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1742?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1726** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1726?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1726?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1725** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1725?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1724** <a href="https://app.graphite.com/github/pr/opengovsg/plumber/1724?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `trunk/auto-flush`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/backend/src/helpers/archival/build-s3-key.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  testRun: boolean
+}): string {
+  const date = new Date(execution.createdAt)
+  const prefix = execution.testRun
+    ? S3_PREFIX_TEST_EXECUTIONS
+    : S3_PREFIX_EXECUTIONS
+  return [
+    prefix,
+    `flow_id=${execution.flowId}`,
+    `year=${date.getUTCFullYear()}`,
```
Comment (ogp-weeloong) [inline packages/backend/src/helpers/archival/build-s3-key.ts]: SGT easier for us since if we ever need to re-hydrate, impacted users will talk about pipe run times in SGT

**Review by ogp-weeloong — APPROVED** [review]
sgt better leh
